File: LeetCode/125. Valid Palindrome/Solution.java
public class Solution {
    public static void main(String[] args) {
        String s = "A man, a plan, a canal: Panama";
        System.out.println(new Solution().isPalindrome(s));
    }
    public boolean isPalindrome(String s) {
        int i = 0, j = s.length() - 1;
        while (i < j) {
            if (!Character.isLetterOrDigit(s.charAt(i))) {
                i++;
            } else if (!Character.isLetterOrDigit(s.charAt(j))) {
                j--;
            } else {
                char c1 = Character.toLowerCase(s.charAt(i));
                char c2 = Character.toLowerCase(s.charAt(j));
                if (c1 != c2) {
                    return false;
                } else {
                    i++;
                    j--;
                }
            }
        }
        return true;
    }

}

File: LeetCode/125. Valid Palindrome/solution.py
class Solution(object):
    def isPalindrome(self, s):
        """
        :type s: str
        :rtype: bool
        """
        i, j = 0, len(s) - 1
        while i < j:
            if not s[i].isalnum():
                i += 1
            elif not s[j].isalnum():
                j -= 1
            elif s[i].lower() == s[j].lower():
                i += 1
                j -= 1
            else:
                return False
        return True

    def isPalindrome1(self, s):
        t = [c.lower() for c in s if c.isalnum()]
        return t == t[::-1]
